src/main/java/competition/subsystems/pose/PoseSubsystem.java

```diff
@@ -13,11 +17,22 @@
 
     final DriveSubsystem drive;
 
+    // odometry
+    final DifferentialDriveOdometry odometry;
+    final DoubleProperty robotX;
+    final DoubleProperty robotY;
+    final DoubleProperty robotHeading;
+
     @Inject
     public PoseSubsystem(CommonLibFactory clf, PropertyFactory propManager, DriveSubsystem drive) {
         super(clf, propManager);
-
         this.drive = drive;
+
+        odometry = new DifferentialDriveOdometry(getRotation2DGyroAngle());
+        robotX = propManager.createEphemeralProperty("robotX", 0.0);
```

Comment:
The robot already has so many X and Y properties that we need to be careful about naming. I recommend calling them something like "wpiPose X", "wpiPose Y".